props/saber_fett263_buttons.h

```diff
@@ -381,160 +1813,2794 @@ SaberFett263Buttons() : PropBase() {}
     if (dynamic_mixer.get_volume() < VOLUME) {
       dynamic_mixer.set_volume(std::min<int>(VOLUME + VOLUME * 0.1,
                                              dynamic_mixer.get_volume() + VOLUME * 0.10));
-      beeper.Beep(0.5, 2000);
+      PlayMenuSound("volup.wav");
       STDOUT.print("Current Volume: ");
       STDOUT.println(dynamic_mixer.get_volume());
     } else {
-      beeper.Beep(0.5, 3000);
+      PlayMenuSound("volmax.wav");
     }
+    return;
   }
 
   void VolumeDown() {
     STDOUT.println("Volume Down");
     if (dynamic_mixer.get_volume() > (0.10 * VOLUME)) {
       dynamic_mixer.set_volume(std::max<int>(VOLUME * 0.1,
                                              dynamic_mixer.get_volume() - VOLUME * 0.10));
-      beeper.Beep(0.5, 2000);
+      PlayMenuSound("voldown.wav");
       STDOUT.print("Current Volume: ");
       STDOUT.println(dynamic_mixer.get_volume());
     } else {
-      beeper.Beep(0.5, 1000);
+      PlayMenuSound("volmin.wav");
     }
+    return;
   }
 
+RefPtr<BufferedWavPlayer> wav_player;
+
+// Menu Sound Player
+void PlayMenuSound(const char* file) {
+  if (!wav_player) {
+  wav_player = GetFreeWavPlayer();
+  if (!wav_player) return;
+  }
+  wav_player->set_volume_now(1.0);
+  if (wav_player->PlayInCurrentDir(file)) return;
+}
+
+  float current_menu_angle_ = 0.0;
+  float current_twist_angle_ = 0.0;
+  float clash_t_ = 2.00;
+#ifdef FETT263_EDIT_MODE_MENU
+  LSPtr<char> restore_point;
+  int blade_length_;
+  int max_length_;
+  float dim;
+  float dim_revert_;
+#endif
+
   bool Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
     switch (EVENTID(button, event, modifiers)) {
-      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_ON):
-      case EVENTID(BUTTON_AUX, EVENT_PRESSED, MODE_ON):
-        if (accel_.x < -0.15) {
-          pointing_down_ = true;
-        } else {
-          pointing_down_ = false;
-        }
-        return true;
-
+      
       case EVENTID(BUTTON_POWER, EVENT_LATCH_ON, MODE_OFF):
       case EVENTID(BUTTON_AUX, EVENT_LATCH_ON, MODE_OFF):
       case EVENTID(BUTTON_AUX2, EVENT_LATCH_ON, MODE_OFF):
-      case EVENTID(BUTTON_POWER, EVENT_CLICK_SHORT, MODE_OFF):
-        if (mode_volume_) {
-          VolumeUp();
+      case EVENTID(BUTTON_POWER, EVENT_CLICK_SHORT, MODE_OFF):                
+        if (menu_) {
```

Comment:
Kind of seems like all this menu exiting code should go in On() / FastOn() so that it also applies to the "on" command in the serial monitor, or when someone clicks "on" in the webusb interface.

Reply:
Entering Edit Mode is done while Off, this code has been moved to MenuChoice() function but doesn't fit with On command, the PWR button in Edit Mode is used for "affirmative" confirmations not ignitions.